File: Web/src/main/java/com/itechart/webflux/web/util/CollectionUtil.java
package com.itechart.webflux.web.util;

import com.itechart.webflux.web.core.model.entity.Entity;

import java.util.ArrayList;
import java.util.List;

public class CollectionUtil {

    public static List<String> getIds(List<? extends Entity> entities) {
        List<String> ids = new ArrayList<>(entities.size());
        for (Entity entity : entities) {
            ids.add(entity.getId());
        }
        return ids;
    }

}

File: Web/src/main/java/com/itechart/webflux/web/util/CollectionUtil.kt
package com.itechart.webflux.web.util

import com.itechart.webflux.web.core.model.entity.Entity
import java.util.ArrayList

class CollectionUtil {

    fun getIds(entities: Iterable<Entity>): List<String> {
        val ids = ArrayList<String>()
        for (entity in entities) {
            entity.id?.let { ids.add(it) }
        }
        return ids
    }

}